lib/index.js

```diff
@@ -133,6 +152,172 @@ iconv.getDecoder = function getDecoder(encoding, options) {
     return decoder;
 }
 
+String.prototype.codePointAt || (String.prototype.codePointAt = function(index) {
+    var code = this.charCodeAt(index);
+
+    if (!isNaN(code) && 0xD800 <= code && code <= 0xDBFF) {
+        var surr = this.charCodeAt(index + 1);
+
+        if (!isNaN(surr) && 0xDC00 <= surr && surr <= 0xDFFF)
+            code = 0x10000 + ((code - 0xD800) << 10) + (surr - 0xDC00);
+    }
+
+    return code;
+});
+
+function TransliterationWrapper(encoder, options) {
+    this.encoder = encoder;
+    this.encodingName = encoder.encodingName + '-translit';
+    this.smartSpacing = (options && options.smartSpacing);
+    this.options = { deferredSmartSpacing: this.smartSpacing, german: options && options.german };
+    this.pending = '';
+}
+
+var MAX_PENDING = 16384;
+
+TransliterationWrapper.prototype.write = function(str) {
+    str = iconv.transliterate(str, this.encoder.encodingName, this.options);
```

Comment:
One question I had is - would this handle surrogate chars? I.e. when a high surrogate is at the end of the previous block and a low surrogate is at the beginning of the current one? It's a valid situation when encoding.

Reply:
Yes, that's taken care of. If the original unidecode is used, it only outputs ASCII, so there are no surrogates to worry about. If unidecode-plus is used, it processes a codepoint at a time, rather than a character at a time, so it keeps the surrogates together.

It is possible for surrogates to be split by `TransliterationWrapper`, in this code which is currently at line 187 of index.js:
```
        // Split the text being written out at a safe place where smart spacing won't
        // need to make any changes.
        var $ = /^(.*)([^\s\x80\x81][\s\x80\x81].*)$/.exec(str);
```
...but the logic works in such a way that they'll always get put back together again properly.